Resources/public/js/views/ez-trashview.js

```diff
@@ -122,6 +125,47 @@ YUI.add('ez-trashview', function (Y) {
         },
 
         /**
+         * Restores the selected (using UDW) trash items
+         *
+         * @protected
+         * @method _restoreSingleTrashItem
+         * @param {EventFacade} e
+         */
+        _restoreSingleTrashItem: function (e) {
+            var trashItemId = e.target.getAttribute('data-trash-item-id');
+
+            e.preventDefault();
+            this.fire('contentDiscover', {
+                config: {
+                    title: Y.eZ.trans('trash.ancestor.to.select', {}, 'trash'),
+                    multiple: false,
+                    contentDiscoveredHandler: Y.bind(function (e) {
+                        var trashItems = [];
+
+                        Y.Array.some(this.get('trashItems'), function (trashItem) {
```

Comment:
I would put that in a dedicated method to avoid duplicating the code here (compared to `_restoreTrashItems`)